gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +207,766 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+            "n_alt_alleles": t.info.n_alt_alleles,
+        },
+        50,
+        140,
+        rf=rf_filter,
+    )
+
+
+def make_field_check_dicts(
+    field_check_expr: dict,
+    field_check_details: dict,
+    check_description: str,
+    cond_expr: hl.expr.BooleanExpression,
+    display_fields: List[str],
+) -> Tuple[dict, dict]:
+    """
+    Create dictionary for aggregating each sanity check's failure count and another for each check's details.
+
+    :param field_check_expr: Dictionary of check description and aggregated expression filtering count
+    :param field_check_details: Dictionary of structs containing each check descriptions details
+    :param check_description: Check to be added to the dictionary
+    :param cond_expr: Logical expression referring to annotations in ht to be checked.
+    :param display_fields: List of ht annotations to be displayed in case of failure (for troubleshooting purposes); these fields are also displayed if verbose is True.
+    :return: Tuple of dictionaries
+    """
+    field_check_expr[check_description] = hl.agg.filter(cond_expr, hl.agg.count())
+    field_check_details[check_description] = hl.struct(
+        cond_expr=cond_expr, display_fields=display_fields
+    )
+
+    return field_check_expr, field_check_details
+
+
+def generic_field_check_loop(
+    ht: hl.Table,
+    field_check_expr: dict,
+    field_check_details: dict,
+    verbose: bool,
+    show_percent_sites: bool = False,
+    ht_count: int = None,
+):
+    """
+    Loop through all conditional checks for a given hail Table.
+
+    This loop allows aggregation across the hail Table once, as opposed to aggregating during every conditional check. 
+    :param ht: Table containing annotations to be checked.
+    :param field_check_expr: Dictionary whose keys are conditions being checked and values are the expressions for filtering to condition.
+    :param field_check_details: Dictionary whose keys are the the check descriptions and values are struct expressions used for check and what to display for the check in the terminal.
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False, show only top values of annotations that fail checks.
+    :param show_percent_sites: Show percentage of sites that fail checks. Default is False.
+    :param ht_count: Previously computed sum of sites within hail Table. 
+    """
+    ht_field_check_counts = ht.aggregate(hl.struct(**field_check_expr))
+    for check_description, n_fail in ht_field_check_counts.items():
+        generic_field_check(
+            ht,
+            cond_expr=field_check_details[check_description].cond_expr,
+            check_description=check_description,
+            n_fail=n_fail,
+            display_fields=field_check_details[check_description].display_fields,
+            verbose=verbose,
+            show_percent_sites=show_percent_sites,
+            ht_count=ht_count,
+        )
```

Comment:
PART OF LARGE SUGGESTION
```suggestion
    ht_field_check_counts = ht.aggregate(
        hl.struct(**{k: v['expr'] for k, v in field_check_expr.items()})
    )
    for check_description, n_fail in ht_field_check_counts.items():
        generic_field_check(
            ht,
            check_description=check_description,
            n_fail=n_fail,
            display_fields=field_check_expr[check_description]['display_fields'],
            verbose=verbose,
            show_percent_sites=show_percent_sites,
            ht_count=ht_count,
        )
```